### PR TITLE
Adding hasFunction/Auto Include PHP Functions Method To Volt Compiler

### PR DESCRIPTION
**Pull request by Fenikkusu**
Hello!

* Type:  new feature
* Link to issue: NA

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:

In many of my projects, I find myself constantly adding PHP Internal Functions to the Volt Compiler Engine (IE: `substr`).

I found this very repetitive and frustrating to have to add to all of them. With this in mind, I've updated the compiler to have a new option of `includePhpFunctions`. When `_compileSource` is ran, it looks for this option and automatically registers all php internal functions within the compiler.

To maintain existing code bases, I've added a `hasFunction` method which I use in the function loop such that it only adds it if it wasn't previously added.

Thanks

### CODE REVIEW

File: phalcon/mvc/view/engine/volt/compiler.zep [hunk 1]
```diff
@@ -2210,8 +2219,23 @@ class Compiler implements InjectionAwareInterface
 				}
 				let this->_autoescape = autoescape;
 			}
+
+            /**
+             * Configures the compiler to register PHP Functions automatically.
+             */
+            if isset options["includePhpFunctions"] {
+                let includePhpFunctions = options["includePhpFunctions"];
```
Comment (Jurigag) [inline phalcon/mvc/view/engine/volt/compiler.zep]: How about make it possible to select which exactly php functions we want to include? Just accept boolean or array maybe?

File: phalcon/mvc/view/engine/volt/compiler.zep [hunk 2]
```diff
@@ -2210,8 +2219,23 @@ class Compiler implements InjectionAwareInterface
...
+                if typeof includePhpFunctions != "boolean" {
+                    throw new Exception("'compileAlways' must be a bool value");
```
Comment (Fenikkusu) [inline phalcon/mvc/view/engine/volt/compiler.zep]: Oops...Error Mistake

File: phalcon/mvc/view/engine/volt/compiler.zep (hunk 1 above)
Comment (Fenikkusu) [inline phalcon/mvc/view/engine/volt/compiler.zep]: Well technically I wouldn't really need to use function_exists. Just loop through the list of functions. If not set from the option, set default to be actual functions. I see no reason as why not...

Comment (Jurigag) [inline phalcon/mvc/view/engine/volt/compiler.zep]: Because someone could provie wrong, non existing function name.

**Comment by Fenikkusu** [comment]
I've added the new updates. You can pass either and array or true. @sergeyklay, a question for you. When running 'function_exists' on 'substr', I kept getting false. After looking at the compiled code, I found that function_exists was being changed to zephir_function_exists. I got around it by using call_user_func instead of function_exists. Does this seem like the valid way to do this, or is it possible that zephir_function_exists wasn't working correctly?

Also, not sure what's going on with failures in Travis.

**Comment by sergeyklay** [comment]
@Fenikkusu We'll double check `zephir_function_exists` asap. 

Cc: @sjinks

**Comment by Jurigag** [comment]
Failures on travis are codeception bug, you will need to cherry pick some commit from @sergeyklay repo

**Comment by sergeyklay** [comment]
https://github.com/phalcon/cphalcon/pull/12861

**Comment by sergeyklay** [comment]
@Fenikkusu  The `zephir_function_exists ` function  has been fix in the latest Zephir

File: phalcon/mvc/view/engine/volt/compiler.zep [hunk 3]
```diff
@@ -2210,8 +2219,23 @@ class Compiler implements InjectionAwareInterface
...
 				let this->_autoescape = autoescape;
 			}
+
+            /**
+             * Configures the compiler to register PHP Functions automatically.
+             */
+            if isset options["includePhpFunctions"] {
+                let includePhpFunctions = options["includePhpFunctions"];
+                if typeof includePhpFunctions != "boolean" && typeof includePhpFunctions != "array" {
+                    throw new Exception("'includePhpFunctions' must be a bool or array value");
```
Comment (sergeyklay) [inline phalcon/mvc/view/engine/volt/compiler.zep]:
```zep
"The 'includePhpFunctions' option must be be either a bool or array"
```

File: phalcon/mvc/view/engine/volt/compiler.zep [hunk 4]
```diff
@@ -2191,9 +2199,10 @@ class Compiler implements InjectionAwareInterface
 	{
 		var currentPath, intermediate, extended,
 			finalCompilation, blocks, extendedBlocks, name, block,
-			blockCompilation, localBlock, compilation, options, autoescape;
+			blockCompilation, localBlock, compilation, options, autoescape, includePhpFunctions;
 
 		let currentPath = this->_currentPath;
+        let includePhpFunctions = false;
```
Comment (sergeyklay) [inline phalcon/mvc/view/engine/volt/compiler.zep]: Could you please you the same indents in the whole file?

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
@Fenikkusu ping

**Pull request closed without merging**